api/src/main/java/org/eclipse/microprofile/config/ConfigValue.java

```diff
@@ -0,0 +1,135 @@
+/*
+ *******************************************************************************
+ * Copyright (c) 2009-2017 Mark Struberg and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ *******************************************************************************/
+package org.eclipse.microprofile.config;
+
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Accessor to a configured value.
+ * It follows a builder pattern.
+ *
+ * Accessing the configured value is finally done via {@link #get()} or {@link #getOptional()}
+ *
+ * @author <a href="mailto:struberg@apache.org">Mark Struberg</a>
+ * @author <a href="mailto:gpetracek@apache.org">Gerhard Petracek</a>
+ * @author <a href="mailto:rsmeral@jboss.org">Ron Smeral</a>
+ */
+public interface ConfigValue<T> {
+
+    /**
+     * Sets the type of the configuration entry to the given class and returns this builder.
+     * The default type of a ConfigValue is {@code String}.
+     *
+     * This will use the registered {@link org.eclipse.microprofile.config.spi.Converter} for the given type.
+     **
+     * @param clazz The target type
+     * @param <N> The target type
+     * @return This builder as a typed ConfigValue
+     */
+    <N> ConfigValue<N> as(Class<N> clazz);
+
+    /**
+     * Specify that a resolved value will get cached for a certain amount of time.
+     * After the time expires the next {@link #get()} will again resolve the value
+     * from the underlying {@link Config}.
+     *
+     * @param value the amount of the TimeUnit to wait
+     * @param timeUnit the TimeUnit for the value
+     *
+     * @return This builder
+     */
+    ConfigValue<T> cacheFor(long value, TimeUnit timeUnit);
```

Comment:
I understand that this method allows defining a business requirement to cache a value for at most value+timeUnit period. This is to prevent longer caching for sensitive data, like blacklists, which should be applied ASAP. However, I'm not sure this can be ensured if config sources implement internal caching. Is there a way to notify all config sources that this value should not be cached for more than configured interval?